### ISSUE TITLE
fix(rules): wai-aria should not require owned elements in empty containers

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3589]
## Describe the bug

The `wai-aria` rule reports violations on empty list/menu/tree containers for not having required owned elements (e.g., `listitem` for `list` role). However, empty containers are valid — the "required owned elements" constraint applies when children are present, not when the container is empty.

## Code Example

```html
<ul></ul>
<div role="list"></div>
<menu></menu>
```

## The current behavior

```
The child element requires the "listitem" role. Or, require aria-busy="true"
```

## The expected behavior

No violations on empty containers. The required owned elements check should only trigger when the container has child elements that don't satisfy the role requirement.

## Impact

Causes ~50 false positives in the nu-html-checker compatibility benchmark.

## Affected package

`@markuplint/rules` — `wai-aria` rule (required owned elements check)

**Comment by YusukeHirao** [comment]
Blocked by #3645 (wai-aria rule split into sub-rules). The empty container check needs its own severity level, which requires the rule to be split.

**Comment by YusukeHirao** [comment]
v5.0 リリースブロッカー（A群）として #3684 に整理。false positive ~50件。
